firmware/lib/Gantry/include/Gantry.h

```diff
@@ -0,0 +1,27 @@
+#ifndef GANTRY_H
+#define GANTRY_H
+
+typedef enum {
+    AXIS_DIR_POSITIVE,
+    AXIS_DIR_NEGATIVE
+} AxisDirection;
+
+typedef enum {
+    AXIS_X,
+    AXIS_Y
+} AxisId;
+
+/**
+ * @enum AxisResult
+ * 
+ * @brief Possible return values for a call to axis_start
+ */
+typedef enum {
+    AXIS_OK,                  //!< Axis movement started OK
+    AXIS_ERR_ALREADY_MOVING,  //!< Axis is already moving
+    AXIS_ERR_LS_HOME,         //!< Trying to move backward while HOME limit switch is pressed
+    AXIS_ERR_LS_FAR,          //!< Trying to move forward while FAR limit switch is pressed
+    AXIS_ERR_INVALID          //!< The parameters resulted in an invalid motion profile
+} AxisResult;
```

Comment:
Is there a message for when the axis has reached it's destination successfully?

Reply:
There is no message, it's up to MIDAS to monitor the response from `GET_STATUS`.
- `MOVING`: Still moving
- `IDLE`: Reached destination
- `FAULT`: Hit a limit switch before reaching destination